### PR TITLE
Add size and ndim to LayerDataProtocol

### PR DESCRIPTION
**Pull request by Carreau**
in `napari/layers/image/image.py`, `_calc_data_range(self...)` calls `calc_data_range(input_data,...)` where `input_data` is a `LayerDataProtocol`

`calc_data_range(...)` needs it to have a size, and ndim. So my guess is we need to make sure that LayerDataProtocol has size and ndim, is that correct ?

Thoughts ? 

Typing all the things that need to be types for this problem to show up is quite long, please see all my other typing PRs to start enabling type checking for these files, mostly opening this separately to know if this is something we wan to add to LayerDataProtocol

### CODE REVIEW

**Review by Czaki — APPROVED** [review]

**Comment by Carreau** [comment]
add the corresponding (`size`, `ndim`) properties to `MultiScaleData`.

File: napari/layers/_data_protocols.py [hunk 1]
```diff
@@ -81,6 +81,14 @@ def __getitem__(
     ) -> LayerDataProtocol:
         """Returns self[key]."""
 
+    @property
+    def size(self) -> int:
+        pass
```
Comment (Czaki) [inline napari/layers/_data_protocols.py]:
```suggestion
        raise NotImplementedError()
```

pass is not excluded from coverage

File: napari/layers/_data_protocols.py [hunk 2]
```diff
@@ -81,6 +81,14 @@ def __getitem__(
...
+
+    @property
+    def ndim(self) -> int:
+        pass
```
Comment (Czaki) [inline napari/layers/_data_protocols.py]:
```suggestion
        raise NotImplementedError()
```

**Review by Czaki — COMMENTED** [review]
Change to `raise NotImplementedError()` or add `pass` to `coverage:report` `exclude_lines` in `setup.cfg`

File: napari/layers/_data_protocols.py (hunk 1 above)
Comment (Carreau) [reply napari/layers/_data_protocols.py]: That is inconsistant with the dtype and getitem, shoudl we:
 - [ ] be consistant and raise notimpl in all of them or,
 - [ ] just leave the body empty with a docstring ?

Comment (Czaki) [reply napari/layers/_data_protocols.py]: I like "just leave ..." option

**Pull request merged**